File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
		Map<Character, Integer> map = new HashMap<>();
		for (char c : stones.toCharArray()) {
			map.put(c, map.getOrDefault(c, 0) + 1);
		}

		int answer = 0;
		for (char c : jewels.toCharArray()) {
			answer += map.getOrDefault(c, 0);
		}

		return answer;
    }
}

File: 0771-jewels-and-stones/0771-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var set: MutableSet<Char> = HashSet()
        var count = 0

        for (j in jewels.toCharArray()) {
            set.add(j)
        }

        for (s in stones.toCharArray()) {
            if (set.contains(s)) count++
        }
        return count
    }
}